ipatests/test_integration/test_adtrust_install.py

```diff
@@ -303,3 +305,591 @@ def test_ipa_user_s4u2self_pac(self):
+        """
+        Test that ipa-adtrust-install install is successful
+        with default rid and secondary values
+        """
+        rid_base = '1000'
+        sec_rid_base = '100000000'
+        self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "--rid-base", rid_base,
+             "--secondary-rid-base", sec_rid_base,
+             "-U"], raiseonerr=False
+        )
+
+    def test_ipa_adtrust_install_with_add_agents_option(self):
+        """
+        This testcase checks that ipa-adtrust-install
+        with --add-agents works without any error
+        on a non-replica IPA server
+        """
+        self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "--add-agents",
+             "-U"], raiseonerr=False
+        )
+
+    def test_ipa_adtrust_install_with_add_sids_option(self):
+        """
+        This testcase checks that ipa-adtrust-install
+        with --add-sids option works without any error
+        """
+        msg = (
+            'adding SIDs to existing users and groups\n'
+            'This step may take considerable amount of time, please wait..'
+        )
+        result = self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "--add-sids",
+             "-U"], raiseonerr=False
+        )
+        assert msg in result.stdout_text
+
+    def test_cldap_responder_doesnot_hang_for_domain_discovery(self):
+        """
+        This testcase checks that cldap responder doesnot hang
+        for domain discovery.
+        https://pagure.io/freeipa/issue/3639
+        """
+        version = tasks.get_openldap_client_version(self.master)
+        if version >= '2.6':
+            pytest.skip('bz2167328')
+        base_dn = str(self.master.domain.basedn)
+        srch_filter = "(&(DnsDomain='{}')(NtVer=\06\00\00\00)" \
+                      "(AAC=\00\00\00\00))".format(base_dn)
+        self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "-U"], raiseonerr=False
+        )
+        result = self.master.run_command(
+            ["ldapsearch", "-LL", "-H",
+             "cldap://{}".format(self.master.hostname),
+             "-b", "", "-s", "base", base_dn, srch_filter]
+        )
+        assert result.returncode == 0
+        assert 'dn: \n netlogon::*' in result.stdout_text
+
+    def test_user_connects_smb_share_if_locked_specific_group(self):
+        """
+        This testcase checks that with ipa-adtrust-installed and
+        the sidgen task registered and run, users cannot connect
+        a share in the local samba server if it is locked to specific
+        groups via "valid users", even if the user is a member of one
+        of those groups.
```

Comment:
This is the description of the issue, not of the test scenario.
Test scenario:
- create a share in the samba server
- access the share as admin, should work
- set valid users = admins to limit the share access to members of the "admins" group
- access the share as admin, should work